### PR TITLE
Add support for the HDBits torrent tracker.

### PR DESCRIPTION
**Pull request by scherz0**
The indexer implementation borrows heavily from the BroadcastTheNet
implementation as HDBits also provides a JSON API that can be used
to query both the recent torrents and the catalog.

### CODE REVIEW

File: src/NzbDrone.Core/Indexers/HDBits/HdBitsApi.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        public uint Comments { get; set; }
+
+        [JsonProperty(PropertyName = "numfiles")]
+        public uint NumFiles { get; set; }
+
+        [JsonProperty(PropertyName = "filename")]
+        public string FileName { get; set; }
+
+        [JsonProperty(PropertyName = "freeleech")]
+        public string FreeLeach { get; set; }
```
Comment (Taloth) [inline src/NzbDrone.Core/Indexers/HDBits/HdBitsApi.cs]: FreeLeech

File: src/NzbDrone.Core/Indexers/HDBits/HdBitsParser.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+            return BuildUrl("https://hdbits.org/download.php", args);
+        }
+
+        private string GetInfoUrl(long torrentId)
+        {
+            var args = new NameValueCollection(1);
+            args["id"] = torrentId.ToString();
+
+            return BuildUrl("https://hdbits.org/details.php", args);
```
Comment (Taloth) [inline src/NzbDrone.Core/Indexers/HDBits/HdBitsParser.cs]: What about use the BaseUrl in the settings? (Appending /api in the RequestGenerator). Allows ppl to change the root url if needed.

**Comment by Taloth** [comment]
Are you sure HDbits uses tvdb numbering instead of scene?

File: src/NzbDrone.Core.Test/IndexerTests/HdBitsTests/HdBitsFixture.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            first.Title.Should().Be("Supernatural S10E17 1080p WEB-DL DD5.1 H.264-ECI");
+            first.DownloadProtocol.Should().Be(DownloadProtocol.Torrent);
+            first.DownloadUrl.Should().Be("https://hdbits.org/download.php?id=257142&passkey=fakekey");
+            first.InfoUrl.Should().Be("https://hdbits.org/details.php?id=257142");
+            first.PublishDate.Should().Be(DateTime.Parse("2015-04-04T20:30:46+0000"));
+            first.Size.Should().Be(1718009717);
+            first.MagnetUrl.Should().BeNullOrEmpty();
+            first.Peers.Should().Be(47);
+            first.Seeders.Should().Be(46);
+        }
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/IndexerTests/HdBitsTests/HdBitsFixture.cs]: Looks to me like infohash is available too in the results.

File: src/NzbDrone.Core/Indexers/HDBits/HdBitsParser.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            }
+
+            var queryResults = responseData.ToObject<TorrentQueryResponse[]>();
+
+            foreach (var result in queryResults)
+            {
+                var id = result.Id;
+                torrentInfos.Add(new TorrentInfo()
+                {
+                    Guid = string.Format("HDB-{0}", id),
```
Comment (Taloth) [inline src/NzbDrone.Core/Indexers/HDBits/HdBitsParser.cs]: Lets make it HDBits-

File: src/NzbDrone.Core.Test/IndexerTests/HdBitsTests/HdBitsFixture.cs (hunk 3 above)
Comment (scherz0) [inline src/NzbDrone.Core.Test/IndexerTests/HdBitsTests/HdBitsFixture.cs]: There's no documentation on what "InfoHash" means in this context.

**Comment by scherz0** [comment]
@Taloth What numbering are you referring to?

**Comment by Taloth** [comment]
Season and Episode numbers. Check out american dad for example.

File: src/NzbDrone.Core.Test/IndexerTests/HdBitsTests/HdBitsFixture.cs (hunk 3 above)
Comment (markus101) [inline src/NzbDrone.Core.Test/IndexerTests/HdBitsTests/HdBitsFixture.cs]: I believe @Taloth is referring to the `hash` property, Sonarr will use that for blacklisting purposes.

Comment (Taloth) [inline src/NzbDrone.Core.Test/IndexerTests/HdBitsTests/HdBitsFixture.cs]: Yeah, I assume that hash property is the torrent infohash.

**Comment by zone117x** [comment]
Whats the status on this? Would love for it to get merged.

**Comment by zone117x** [comment]
@Taloth I can confirm HDBits uses normal tvdb numbering. American Dad for example, the API query is: 
`
  "tvdb": { 
    "id": 73141,
    "season": 12,
    "episode": 10
  }
`

Looks like this pull request works just fine.

**Comment by Taloth** [comment]
Merged into develop.

**Comment by intelligence** [comment]
When will it be added to main?

**Comment by markus101** [comment]
@intelligence This was merged months ago, make sure Sonarr is up to date...

**Pull request closed without merging**